src/types/dict.rs

```diff
@@ -228,51 +228,129 @@ impl PyDict {
     }
 }
 
-pub struct PyDictIterator<'py> {
-    dict: &'py PyAny,
-    pos: isize,
-}
+#[cfg(all(Py_3_8, not(PyPy), not(Py_LIMITED_API)))]
+mod impl_ {
+    use super::*;
+    use std::marker::PhantomData;
+
+    pub struct PyDictIterator<'py> {
+        di_dict: *mut ffi::PyDictObject,
+        di_pos: ffi::Py_ssize_t,
+        di_used: ffi::Py_ssize_t,
+        marker: PhantomData<&'py PyDict>,
```

Comment:
Is there a risk that the dict refcount can fall to 0 and can be accidentally released? Should this be `Py<PyDict>`?

Reply:
The iterator borrows from the Pydict, so this should be fine.